SEO Tools: add archive_title page title token

samiff opened the pull request:
#### Changes proposed in this Pull Request

Related (prerequisite): https://github.com/Automattic/jetpack/pull/20920

This PR:

* Supports adding & previewing a more generic `archive_title` token for custom Archive page titles.
* Hides the `Date` token button for Archives (but we still tokenize existing [date] usage).
* The `Date` button will only be replaced with the `Archive Title` button on Jetpack versions `>= 10.2-alpha`

#### Testing instructions

* See: https://github.com/Automattic/jetpack/pull/20920

---

Screenshot for Translators:

![Markup on 2021-09-10 at 13:33:55](https://user-images.githubusercontent.com/15803018/132907722-5dfe80ff-f14f-4c7b-abba-8f20301e3a39.png)

samiff committed:
Add archiveTitle token

samiff committed:
Hide date insertion button for Archives title

samiff committed:
Update date/archiveTitle preview

a8ci18n commented on client/components/seo/meta-title-editor/index.jsx:
```diff
@@ -23,14 +23,15 @@ const getValidTokens = ( translate ) => ( {
 	pageTitle: translate( 'Page Title' ),
 	groupTitle: translate( 'Tag or Category Name' ),
 	date: translate( 'Date' ),
+	archiveTitle: translate( 'Archive Title' ),
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **21** times:
`translate( 'Testimonial Archive Title' )` *ES Score: 6*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/seo-archive-title)

samiff committed:
Update test

samiff committed:
Add version check for JP >= 10.2

Otherwise, a WPorg JP site on say 10.1 would see the option to add an
Archive Title in Calypso, but it would fail at saving it.

samiff committed:
Merge branch 'trunk' into add/seo-archive-title

samiff commented:
@jeherve Does the version check added in https://github.com/Automattic/wp-calypso/pull/56084/commits/e4e21faeb74fd2484b0aca4c06f23167fdad92dc look ok?

jeherve reviewed (commented):
> Does the version check added in e4e21fa look ok?

Yup. Although you could use the `isJetpackMinimumVersion` utility, to have less work to do?

https://github.com/Automattic/wp-calypso/blob/7655b6d1f326371990851da10ebb97f133fbdc17/client/state/sites/selectors/is-jetpack-minimum-version.js#L5-L15

samiff committed:
Use isJetpackMinimumVersion to check version

samiff commented:
> Although you could use the `isJetpackMinimumVersion` utility, to have less work to do?

Thanks! Totally missed that one.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6581579

Hi @samiff, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `Archive Title`
- `Example Archive Title/Date`

Thank you in advance!

samiff commented:
@jeherve Small snag here, `isJetpackMinimumVersion` or fetching `getSiteOption( state, siteId, 'jetpack_version' )` won't return truthy for a WPcom non-AT Business plan site which should have the SEO settings.

Would this work, or is there a better way?

```js
	let shouldShowSeoArchiveTitleButton = false;
	if ( isJetpackMinimumVersion( state, siteId, '10.2-alpha' ) ) {
		shouldShowSeoArchiveTitleButton = true;
	} else if (
		! isJetpackSite( state, siteId ) &&
		hasActiveSiteFeature( state, siteId, FEATURE_ADVANCED_SEO )
	) {
		// For non-AT Business plan sites which get SEO features.
		shouldShowSeoArchiveTitleButton = true;
	}
```

jeherve commented:
Yeah, that sounds good to me.
